crypto/x509/x509_lu.c

```diff
@@ -324,9 +331,16 @@ static int ossl_x509_store_ctx_get_by_subject(const X509_STORE_CTX *ctx,
     stmp.type = X509_LU_NONE;
     stmp.data.ptr = NULL;
 
-    if (!X509_STORE_lock(store))
+    if (!x509_store_lock_ex(store, 1))
         return 0;
-    sk_X509_OBJECT_sort(store->objs);
+    /* Should already be sorted...but just in case */
+    if (!sk_X509_OBJECT_is_sorted(store->objs)) {
+        X509_STORE_unlock(store);
+        /* Take a write lock instead of a read lock */
+        X509_STORE_lock(store);
+        if (!sk_X509_OBJECT_is_sorted(store->objs))
```

Comment:
Checking this again isn't strictly required.  The sort method checks and exits early if there is nothing to be done.